react-accordion-component/my-app/src/AccordionComponent.jsx

```diff
@@ -0,0 +1,42 @@
+import './Styles.css';
+import { useState } from 'react';
+
+// parent component w/ topics array as a prop
+export default function AccordionComponent({ topics }) {
+  // initializes useState
+  const [chosenTopic, setChosenTopic] = useState();
+
+  // toggles topic based on index - if open topic is clicked - will close, if a closed topic is clicked - will open
+  const handleClick = (index) => {
+    if (chosenTopic === index) {
+      setChosenTopic(null);
+    } else {
+      setChosenTopic(index);
+    }
+  };
+
+  // creates list of Topic components by mapping thru topics array and creates a Topic for each w/ its properties as props
+  const topicComponents = topics.map((topic, index) => (
+    <Topic
+      key={topic.id}
+      title={topic.title}
+      content={topic.content}
+      isOpen={chosenTopic === index}
+      handleClick={() => handleClick(index)}
+    />
+  ));
+  // renders Topic components in a div
+  return <div>{topicComponents}</div>;
+}
+
+// child component - contains button w/ title and <p> with text
+function Topic({ title, content, isOpen, handleClick }) {
```

Comment:
Instead of `handleClick` it would be more idiomatic React to name it `onClick` -- the prop is named with `on`, the function is named with `handle`.